instrumentation-genai/opentelemetry-instrumentation-vertexai/src/opentelemetry/instrumentation/vertexai/utils.py

```diff
@@ -208,3 +257,26 @@ def _parts_to_any_value(
         cast("dict[str, AnyValue]", type(part).to_dict(part))  # type: ignore[reportUnknownMemberType]
         for part in parts
     ]
+
+
+def _map_finish_reason(
+    finish_reason: content.Candidate.FinishReason
+    | content_v1beta1.Candidate.FinishReason,
+) -> FinishReason | str:
+    EnumType = type(finish_reason)  # pylint: disable=invalid-name
+    if (
+        finish_reason is EnumType.FINISH_REASON_UNSPECIFIED
+        or finish_reason is EnumType.OTHER
+    ):
+        return "error"
+    if finish_reason is EnumType.STOP:
+        return "stop"
+    if finish_reason is EnumType.MAX_TOKENS:
+        return "length"
+
+    # There are a lot of specific enum values from Vertex that would map to "content_filter".
```

Comment:
Is `content_filter` something you came up with to generalize the other finish reasons or is this defined by OT somewhere?

Reply:
It's defined here https://github.com/open-telemetry/semantic-conventions/blob/v1.30.0/docs/gen-ai/gen-ai-events.md#:~:text=finish_reason%20has%20the%20following%20list%20of%20well%2Dknown%20values.%20If%20one%20of%20them%20applies%2C%20then%20the%20respective%20value%20MUST%20be%20used%3B%20otherwise%2C%20a%20custom%20value%20MAY%20be%20used.